index.js

```diff
@@ -335,10 +338,11 @@
   var TypeRep = $.NullaryType(
     'sanctuary/TypeRep',
     function(x) {
-      return R.type(x) === 'Function' ||
-             (x != null &&
-              R.type(x.name) === 'String' &&
-              R.type(x.test) === 'Function');
+      return R.type(x) === 'Function'
+          && R.type(global_[x.name]) === 'Function'
+          || x != null
+          && x.prototype != null
+          && R.type(x.prototype['@@type']) === 'String';
```

Comment:
Good point. Let's reinstate the `R.type(x) === 'Function'` predicate, so we continue to treat *any* function as a type representative. This will obviate the need for `R.type(global_[x.name]) === 'Function'` and a way to locate the global object, so I'm very much in favour!